packages/SwingSet/src/kernel/kernel.js

```diff
@@ -306,7 +320,7 @@ export default function buildKernel(
     insistCapData(info);
     // if vatFatalSyscall was here already, don't override: bad syscalls win
     if (!terminationTrigger) {
-      terminationTrigger = { vatID, abortCrank: false, reject, info };
+      terminationTrigger = { vatID, abortCrank: reject, reject, info };
```

Comment:
huh, so we were allowing messages from the crank that did `vatPowers.exitVatWithFailure()`? oops.

I wonder what I was thinking. I'd say we should update the `vatPowers` docs to make it clear that `exitVat` accepts the vat's final messages, and `exitVatWithFailure` unwinds them, but apparently `vatPowers` doesn't show up in `packages/SwingSet/docs/` anywhere, which is too bad. I'm pretty sure @erights is depending upon it working this way.

Thanks for fixing this and updating the test to catch it.